python/lsst/dbb/buffmngrs/endpoint/backfill/backfill.py

```diff
@@ -0,0 +1,190 @@
+                    acc["failure"] += 1
+                    continue
+                if record is not None:
+                    logger.warning(f"file already tracked (id: {record.id})")
+                    logger.debug(f"{relpath}: terminating processing")
+                    acc["failure"] += 1
+                    continue
+
+                logger.debug(f"calculating checksum")
+                try:
+                    checksum = get_checksum(abspath)
+                except FileNotFoundError:
+                    logger.error(f"{relpath}: no such file")
+                    logger.debug(f"{relpath}: terminating processing")
+                    acc["failure"] += 1
+                    continue
+
+                # Always make BOTH inserts in a single transaction!
+                # Otherwise new entry in file table may be picked up by
+                # an Ingester daemon running in the background.
+                logger.debug(f"updating database entries")
+
+                # Add file record (starts the transaction).
+                file = self.File(
+                    relpath=dirname,
+                    filename=filename,
+                    checksum=checksum,
+                )
+                self.session.add(file)
+
+                # Flush the changes to get the id for that record (does NOT
+                # end the transaction).
+                try:
+                    self.session.flush()
+                except SQLAlchemyError as ex:
+                    self.session.rollback()
+                    logger.error(f"{relpath}: cannot create file entry: {ex}")
+                    logger.debug(f"{relpath}: terminating processing")
+                    acc["failure"] += 1
+                    continue
+
+                # Add the corresponding event record.
+                event = self.Event(
+                    status=Status.BACKFILL.value,
+                    start_time=datetime.datetime.now(),
+                    files_id=file.id,
+                )
+                self.session.add(event)
+
+                # Finally, commit the changes or roll the changes back if
+                # any errors were encountered (ends the transaction).
+                try:
+                    self.session.commit()
+                except Exception as ex:
+                    self.session.rollback()
+                    logger.error(f"{relpath}: cannot create database entries: "
+                                 f"{ex}")
+                    acc["failure"] += 1
+                else:
+                    logger.debug(f"{relpath}: processing completed")
+                    acc["success"] += 1
+        total = sum(x for x in acc.values())
+        if total == 0:
+            logger.warning("no backfill attempts were made")
+        if acc["failure"] != 0:
+            logger.warning("some backfill attempts failed")
+        else:
+            logger.info("all found files were backfilled successfully")
```

Comment:
Since the code has the count of successes why not print it?

No particular reason, I just assumed no one would care how many files were backfilled successfully _if_ there were no failures (of any kind).

However, in the context your earlier suggestion about distinguishing between different types of failed backfills (DB errors vs. already tracked), now it will make sense to presents these statistics. I will make the necessary changes.